src/app/controllers/LoginController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\controllers;
+
+use App\Core\Controller;
+use App\core\Responses\HTMLResponse;
+use App\core\Responses\IResponse;
+use App\models\LoginModel;
+
+class LoginController extends Controller
+{
+    public function login($params): IResponse
+    {
+        $data = [...$params];
+
+        if ((isset($_SESSION['user'], $_SESSION['email']))) {
+            $data['user'] = ['email' => $_SESSION['email'], 'name' => $_SESSION['user']];
+        }
+
+        $body = $this->view->render('login', $data);
+
+        return new HTMLResponse(['200 OK'], $body);
+    }
+
+    public function loginPost($params): IResponse
```

Comment:
Could: Too much business logic in controller action. Part of it may be put into LoginModel. Try to keep you controller below 20 lines of code